pandas/tests/test_join.py

```diff
@@ -233,3 +233,22 @@ def test_merge_join_categorical_multiindex():
     result = a.join(b, on=['Cat1', 'Int1'])
     expected = expected.drop(['Cat', 'Int'], axis=1)
     assert_frame_equal(expected, result)
+
+
+def test_join_multi_to_multi():
+    leftindex = MultiIndex.from_product([list('abc'), list('xy'), [1, 2]],
```

Comment:
can you add this issue number here

ideally can you use the join_type fixture here (to run this for all join types)